cmake/environmentchecks.cmake

```diff
@@ -12,12 +12,16 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.cmake_minimum_required (VERSION 3.1)
 
+include(CheckCXXCompilerFlag)
+
 get_property(IS_MULTI_CONFIG GLOBAL PROPERTY GENERATOR_IS_MULTI_CONFIG)
 
 find_package(Python ${PREFERRED_PYTHON_VERSION} COMPONENTS Interpreter Development)
 
 find_program(CLANG_FORMAT_EXE NAMES ${PREFERRED_CLANG_FORMAT_NAMES} clang-format)
 find_program(CLANG_TIDY_EXE NAMES ${PREFERRED_CLANG_TIDY_NAMES} clang-tidy)
+find_program(LLVM_PROFDATA_EXE NAMES ${PREFERRED_LLVM_PROFDATA_NAMES} llvm-profdata HINTS /Library/Developer/CommandLineTools/usr/bin/)
+find_program(LLVM_COV_EXE NAMES ${PREFERRED_LLVM_COV_NAMES} llvm-cov HINTS /Library/Developer/CommandLineTools/usr/bin/)
```

Comment:
1. The programs to be found are currently listed alphabetically, so it might be best to keep it that way?
2. Are the hints really needed? Also, they seem to be tageted at macOS? What about Windows and Linux?

Reply:
1. Ok
2. Yes. I haven't tested on Windows and Linux that have llvm, a case of leave it until actually required.  I was thinking about adding in a variable to add to the hints so a user could add a configuration parameter that would enable finding of llvm in non-standard locations.

Reply:
I have LLVM installed on my Windows, Linux and macOS machines, so I guess I will eventually find out how that behaves... :)

Reply:
Excellent.